doc/source/whatsnew/v0.24.0.txt

```diff
@@ -317,14 +317,34 @@ and respect calendar day arithmetic while :class:`Day` and frequency alias ``'D'
 will now respect absolute time (:issue:`22274`, :issue:`20596`, :issue:`16980`, :issue:`8774`)
 See the :ref:`documentation here <timeseries.dayvscalendarday>` for more information.
 
-Addition with :class:`CalendarDay` across a daylight savings time transition:
+The difference between :class:`Day` vs :class:`CalendarDay` is most apparent
+with timezone-aware datetime data with a daylight savings time transition:
```

Comment:
You say "most apparent with timezone-aware data". But is there a case where this is actually relevant for tz naive data?